// lwc-webpack-plugin/dist/loader.js
"use strict";
exports.__esModule = true;
/*
 * Copyright (c) 2022, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */
var compiler = require('@lwc/compiler');
var getInfoFromPath = require('./module').getInfoFromPath;
var loader_utils_1 = require("loader-utils");
module.exports = function (source) {
    var resourcePath = this.resourcePath;
    var _a = loader_utils_1.getOptions(this), stylesheetConfig = _a.stylesheetConfig, outputConfig = _a.outputConfig, experimentalDynamicComponent = _a.experimentalDynamicComponent;
    var info;
    try {
        info = getInfoFromPath(resourcePath, process.cwd());
    }
    catch (e) {
        info = {
            name: '',
            namespace: ''
        };
    }
    var codeTransformed = source;
    var cb = this.async();
    compiler
        .transform(codeTransformed, resourcePath, {
        name: info.name,
        namespace: info.ns,
        stylesheetConfig: stylesheetConfig,
        outputConfig: outputConfig,
        experimentalDynamicComponent: experimentalDynamicComponent,
        scopedStyles: resourcePath.endsWith('.scoped.css')
    })
        .then(function (res) {
        cb(null, res.code);
    })["catch"](function (err) {
        cb(err);
    });
    return;
};
